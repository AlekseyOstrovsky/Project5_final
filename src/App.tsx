import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import AdminPage from './components/Admin/AdminPage';
import Header from './components/Header';
import Footer from './components/Footer';
import Shop from './components//Shop/Shop';
import ProductPage from './components/Shop/ProductPage';
import Cart from './components/Cart/Cart';

function App() {	
	return (
		<BrowserRouter basename={process.env.PUBLIC_URL}>
			<div>
				<Header/>
				<div className="wrapper container">
					<Routes>
						<Route path='/admin' element={<AdminPage/>}/>
						<Route path='/shop' element={<Shop/>}/>
						<Route index element={<Shop/>}/>
						<Route path='/shop/:id' element = {<ProductPage/>}/>
						<Route path='/cart' element={<Cart/>}/>
					</Routes>
				</div>
				<Footer/>
			</div>
		</BrowserRouter>
	);
}

export default App;
