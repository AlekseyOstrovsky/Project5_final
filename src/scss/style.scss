@import "_nullstyle";

@import "_header";
@import "_footer";
@import "_shop";
@import "_shop-sidebar";
@import "_product-page";
@import "_cart";
@import "_admin-page";


@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800&display=swap');


input[type="number"]::-webkit-outer-spin-button,
input[type="number"]::-webkit-inner-spin-button {
	display: none;
}


body {
	font-family: 'Inter', sans-serif;
	background-color: #FFFFFF;
}

.wrapper {
	min-height: 80vh;
}

.container {	
	margin: 0 auto;
	max-width: 1366px;
}



.route {
	margin-left: -10px;	
	cursor: pointer;
	display: flex;
	font-size: 14px;
	line-height: 120%;
	color: #3F4E65;
	& div {
		padding: 0 10px;
		border-right: dashed 1px rgba(63, 78, 101, 0.3);
	}
	& span {
		padding-left: 10px;
		color:rgba(63, 78, 101, 0.5)
	}
	@media (max-width: 768px) {
		display: none;
	}
}

.route-mobile {
	@media (min-width: 769px) {
		display: none;
	}
	display: flex;
	align-items: center;
	&__button {	
		margin-right: 5px;
		display: flex;
		justify-content: center;
		align-items: center;
		width: 32px;
		height: 32px;
		border-radius: 50%;	
		background-color: #f9dba8;
	}
	&__text {
		font-size: 10px;
		font-weight: 600;
		letter-spacing: 1px;
		color: #3F4E65;
	}
}
