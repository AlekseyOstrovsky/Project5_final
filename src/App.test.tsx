import '@testing-library/jest-dom';
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import App from "./App";
import './css/style.css';

describe('React App component', () => {

	beforeEach( () => {
		window.localStorage.clear()
	})

	afterEach( () => {
		window.localStorage.clear()
	})

	test('contains header and footer', () => {
		render(<App/>);
		expect(screen.getByTestId('header-container')).toBeInTheDocument();	
		expect(screen.getByTestId('footer')).toBeInTheDocument();		
	})

	
	test('contains sidebar', () => {
		render(<App/>);
		expect(screen.getByTestId('sidebar')).toBeInTheDocument();
		expect(screen.getByTestId('sidebar')).toMatchSnapshot();
	})

	
	test('contains usagebar', () => {
		render(<App/>);
		expect(screen.getByTestId('usagebar')).toBeInTheDocument();
		expect(screen.getByTestId('usagebar')).toMatchSnapshot();
	})


	test('contains no goods on first start', () => {
		render(<App/>);
		expect(screen.getByText('Товары отсутствуют')).toBeInTheDocument();				
	})

	test('adds default goods via admin page', () => {
		render(<App/>);
		expect(screen.getByText('Товары отсутствуют')).toBeInTheDocument();
		fireEvent.click(screen.getByTestId('admin'));
		fireEvent.click(screen.getByText('Загрузить товары по умолчанию'));
		fireEvent.click(screen.getByText('Главная'));
		expect(screen.getByText('Крем для рук Увлажнение и мягкость Kora 100 мл')).toBeInTheDocument();
		expect(screen.getByText('0000007259266')).toBeInTheDocument();
		expect(screen.getByText('0000010126600')).toBeInTheDocument();
		expect(screen.getByText('0000010486809')).toBeInTheDocument();
		expect(screen.getByTestId('pagination')).toBeInTheDocument();
		expect(screen.queryByText('Товары отсутствуют')).not.toBeInTheDocument();
	})

	test('adds goods to cart via shop page', () => {
		render(<App/>);	
		fireEvent.click(screen.getByTestId('admin'));
		fireEvent.click(screen.getByText('Загрузить товары по умолчанию'));
		fireEvent.click(screen.getByText('Главная'));
		expect(window.localStorage.getItem('cart')).toBe(null);
		fireEvent.click(screen.queryAllByText('В КОРЗИНУ')[0]);
		expect(window.localStorage.getItem('cart')).not.toBe(null);		
	})

	test('filters goods by usage', () => {
		render(<App/>);	
		fireEvent.click(screen.getByTestId('admin'));
		fireEvent.click(screen.getByText('Загрузить товары по умолчанию'));
		fireEvent.click(screen.getByText('Главная'));
		expect(screen.queryByText('Товары отсутствуют')).not.toBeInTheDocument();
		fireEvent.click(screen.queryAllByText('Уход за ногами')[0]);
		expect(screen.queryByText('Товары отсутствуют')).toBeInTheDocument();
	})

	test('filters goods by price', () => {
		render(<App/>);	
		fireEvent.click(screen.getByTestId('admin'));
		fireEvent.click(screen.getByText('Загрузить товары по умолчанию'));
		fireEvent.click(screen.getByText('Главная'));		
		expect(screen.queryByText('Товары отсутствуют')).not.toBeInTheDocument();
		expect(screen.getByTestId('pagination')).toBeInTheDocument();
		fireEvent.input(screen.getByTestId('maxprice'),{target: {value: '400'}});	
		expect(screen.queryAllByTestId('shopitem').length).toBe(10);
		fireEvent.input(screen.getByTestId('minprice'),{target: {value: '300'}});
		expect(screen.queryAllByTestId('shopitem').length).toBe(6);
	})

	test('opens good page and adds good to cart', () => {
		render(<App/>);	
		fireEvent.click(screen.getByTestId('admin'));
		fireEvent.click(screen.getByText('Загрузить товары по умолчанию'));
		fireEvent.click(screen.getByText('Главная'));
		fireEvent.click(screen.queryAllByTestId('shopitem-name')[0]);
		expect(window.localStorage.getItem('cart')).toBe(null);
		fireEvent.click(screen.getByText('В корзину'));
		expect(window.localStorage.getItem('cart')).not.toBe(null);		
	})

})
