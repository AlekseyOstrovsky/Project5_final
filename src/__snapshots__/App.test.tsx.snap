// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React App component contains sidebar 1`] = `
<div
  class="sidebar"
  data-testid="sidebar"
>
  <div
    class="sidebar__title"
  >
    <span>
      ПОДБОР ПО ПАРАМЕТРАМ
    </span>
    <div
      class="sidebar__mobile-dropdown"
    >
      <img
        alt="Назад"
        src="arrow-mobile.png"
      />
    </div>
  </div>
  <div
    class="sidebar__mobile-dropdown-wrapper"
  >
    <div
      class="price-filter"
    >
      <div
        class="price-filter__header"
      >
        Цена 
        <span>
          ₸
        </span>
      </div>
      <div
        class="price-filter__main"
      >
        <input
          data-testid="minprice"
          type="number"
          value="0"
        />
        <span>
          -
        </span>
        <input
          data-testid="maxprice"
          type="number"
          value="10000"
        />
      </div>
    </div>
    <div
      class="producers__wraper"
    >
      <div
        class="sidebar__title"
      >
        Производитель
      </div>
      <div
        class="producers__input"
      >
        <input
          placeholder="Поиск..."
          type="text"
          value=""
        />
        <div
          class="producers__input-search-logo"
        >
          <img
            alt="Поиск"
            src="search.png"
          />
        </div>
      </div>
    </div>
    <hr
      class="sidebar__hr"
    />
    <div
      class="sidebar__usage"
    >
      <div
        class="sidebar__usage__element"
      >
        Уход за телом
      </div>
      <div
        class="sidebar__usage__element"
      >
        Уход за руками
      </div>
      <div
        class="sidebar__usage__element"
      >
        Уход за ногами
      </div>
      <div
        class="sidebar__usage__element"
      >
        Уход за лицом
      </div>
      <div
        class="sidebar__usage__element"
      >
        Уход за волосами
      </div>
      <div
        class="sidebar__usage__element"
      >
        Средства для загара
      </div>
      <div
        class="sidebar__usage__element"
      >
        Средства для бритья
      </div>
      <div
        class="sidebar__usage__element"
      >
        Подарочные наборы
      </div>
      <div
        class="sidebar__usage__element"
      >
        Гигиеническая продукция
      </div>
      <div
        class="sidebar__usage__element"
      >
        Гигиена полости рта
      </div>
      <div
        class="sidebar__usage__element"
      >
        Бумажная продукция
      </div>
    </div>
  </div>
</div>
`;

exports[`React App component contains usagebar 1`] = `
<div
  class="usage"
  data-testid="usagebar"
>
  <div
    class="usage__element"
  >
    Уход за телом
  </div>
  <div
    class="usage__element"
  >
    Уход за руками
  </div>
  <div
    class="usage__element"
  >
    Уход за ногами
  </div>
  <div
    class="usage__element"
  >
    Уход за лицом
  </div>
  <div
    class="usage__element"
  >
    Уход за волосами
  </div>
  <div
    class="usage__element"
  >
    Средства для загара
  </div>
  <div
    class="usage__element"
  >
    Средства для бритья
  </div>
  <div
    class="usage__element"
  >
    Подарочные наборы
  </div>
  <div
    class="usage__element"
  >
    Гигиеническая продукция
  </div>
  <div
    class="usage__element"
  >
    Гигиена полости рта
  </div>
  <div
    class="usage__element"
  >
    Бумажная продукция
  </div>
</div>
`;
